### DISCUSSION TITLE
tutorial on importing and using npm code?

### DISCUSSION THREAD

**Discussion by pleasantone** [hugomods/.github#4] [Q&A]
Just curious if there is a good tutorial on importing and using npm based js and css in my hbstack based site. Apologies in advance if this is more relevant to hbstack but I thought it general.

or GitHub hosted js/scss code…

looks like an index.ts in assets/modules/____/js/index.ts
but not sure where the npm code goes..?

in any case, a tutorial for dummies as a blog article would be nice…

**Comment by razonyang** [comment]
Will do if I find sometime. Here are some hints may help you getting forward.

1. `npm install -D --save PACKAGE_NAME` to install packages.
2. Check the files of package, e.g. `tree node_modules/foo`, let's say it look like:

```sh
$ tree node_modules/foo
scss/
  index.scss
js/
  index.js
```

## import a JS.

```ts
assets/modules/____/js/index.ts
import "foo/js/index";
```

## import a SASS.

```scss
assets/modules/____/scss/index.scss
@import "../../../foo/scss/index";
```

The path is relative to the `assets`, Hugo will resolve it from `node_modules` if not found in `assets`.

**Comment by pleasantone** [comment]
Oh simple!!! Thanks!
